doc/nrf/libraries/bluetooth_services/mesh/light_hue_srv.rst

```diff
@@ -13,12 +13,18 @@ It should be instantiated in the light fixture node.
 
-Unlike the other Lightness models, the Hue Server model's Hue state will wrap around when reaching its maximum or minimum values as defined in :c:member:`bt_mesh_light_hue_srv.range`.
-To be able to handle this effectively in the application, the Light Hue Server model exposes the underlying Generic Level Server's Move and Delta Set messages, which must be implemented.
-If the application gets a call to the :c:member:`bt_mesh_light_hue_srv_handlers.delta_set` callback, it must move the Hue relative to its current value, in the direction determined by the ``delta_set`` message.
-Note that this may cause the Hue state to wrap around.
-Similarly, if the application gets a call to the :c:member:`bt_mesh_light_hue_srv_handlers.move_set` callback, it must start a continuous transition of the Hue state in the direction and rate determined by the ``move_set`` message.
-This move transition continues until the Hue Server receives a new ``set``, ``delta_set`` or ``move_set`` message, or the device reboots.
+The Hue Server model's Hue state has special rules about wrap around behavior which needs to be considered when writing the application.
+Unlike the other Lightness models, the Hue state's change structure :c:struct:`bt_mesh_light_hue` has additional field, :c:member:`bt_mesh_light_hue.direction`, which tells whether the current Hue state should increase or decrease in order to reach the target value, when the Hue state transition is not instantaneous.
+When the Hue state is not limited by the Hue Range state (the minimum value of the Hue range is ``0`` and the maximum value is ``65535``), the Hue state can wrap around.
+This, for example, can happen when the target value is less than the current one, but the Hue state should increment as claimed by the :c:member:`bt_mesh_light_hue.direction` field.
+This is because the :c:member:`bt_mesh_light_hue.direction` field points to the shortest path towards the target value.
+The Hue state also wraps around when the minimum value of the Hue state is greater than the maximum value.
+In this case, the state wraps around when the value reaches the end of the type range.
+This happens because the :c:member:`bt_mesh_light_hue.direction` field points to the direction of the allowed range of the Hue state values.
+
+When the application gets a call to the :c:member:`bt_mesh_light_hue_srv_handlers.set` callback, the transition always stops at the target value represented by the :c:member:`bt_mesh_light_hue.lvl` field.
+But, if the application gets a call to the :c:member:`bt_mesh_light_hue_srv_handlers.move_set` callback, it must start a continuous transition of the Hue state in the direction and rate determined by the :c:struct:`bt_mesh_light_hue_move` structure.
+This move transition continues until the Hue Server receives a new ``set`` or ``move_set`` message, or the device reboots.
 
 States
```

Comment:
Hue Server -> Light Hue Server
in the note further down.
Not sure if there are more, I don't see them at least :)